### PR TITLE
 Add Groestlcoin (GRS) support 

### PR DESCRIPTION
**Pull request by yura-pakhuchiy**
Tests are in https://github.com/keepkey/python-keepkey/pull/57

Also this PR includes commit which fixes curve name for Decred.

### CODE REVIEW

File: include/keepkey/firmware/coins.def [hunk 1]
```diff
@@ -24,10 +24,12 @@ X(true, "Bitcloud",     true, "BTDX",  true,  25,    true,    1000000,  true,
...
 X(true, "Ravencoin",    true, "RVN",   true,  60,    true,    2000000,  true, 122,  true, "\x1a" "Ravencoin Signed Message:\n",     true, 0x800000af,   false,   0,     true,   8,  false, NO_CONTRACT,      true, 76067358,  true, false,  true, false,  true, SECP256K1_STRING,  false, "",            false, "",     false, false,  false,        0,        false,       0)
-X(true, "Decred",       true, "DCR",   true,1855,    true,    1000000,  true,1818,  true, "\x17" "Decred Signed Message:\n",        true, 0x8000002a,   false,   0,     true,   8,  false, NO_CONTRACT,      true, 50178342,  true, false,  true, false,  true,"secp256k1_decred", false, "",            false, "",      true,  true,  false,        0,        false,       0)
+X(true, "Decred",       true, "DCR",   true,1855,    true,    1000000,  true,1818,  true, "\x17" "Decred Signed Message:\n",        true, 0x8000002a,   false,   0,     true,   8,  false, NO_CONTRACT,      true, 50178342,  true, false,  true, false,  true,"secp256k1-decred", false, "",            false, "",      true,  true,  false,        0,        false,       0)
```
Comment (keepkeyjon) [inline include/keepkey/firmware/coins.def]: Oh, good catch!

File: include/keepkey/firmware/coins.def [hunk 2]
```diff
@@ -24,10 +24,12 @@ X(true, "Bitcloud",     true, "BTDX",  true,  25,    true,    1000000,  true,
...
 X(true, "Monacoin",     true, "MONA",  true,  50,    true,    5000000,  true,  55,  true, "\x19" "Monacoin Signed Message:\n",      true, 0x80000016,   false,   0,     true,   8,  false, NO_CONTRACT,      true, 76067358,  true, true,   true, false,  true, SECP256K1_STRING,  false, "",            true, "mona",  false, false,  true,  77429938,        false,       0)
 X(true, "Electra",      true, "ECA",   true,  33,    true,      10000,  true,  40,  true, "\x1d" "Electra very Signed Message:\n",  true, 0x800000f9,   false,   0,     true,   8,  false, NO_CONTRACT,      true, 76067358,  false, false, false, false, true, SECP256K1_STRING,  false, "",            false, "",     false, false,  false,        0,        false,       0)
+X(true, "Groestlcoin",  true, "GRS",   true,  36,    true,     100000,  true,   5,  true, "\x1c" "GroestlCoin Signed Message:\n",   true, 0x80000011,   false,   0,     true,   8,  false, NO_CONTRACT,      true, 76067358,  true, true,   true, false,  true,"secp256k1-groestl", false, "",           true, "grs",  false, false,  true,  77429938,        true, 78792518)
```
Comment (keepkeyjon) [inline include/keepkey/firmware/coins.def]:
```suggestion
X(true, "Groestlcoin",  true, "GRS",   true,  36,    true,     100000,  true,   5,  true, "\x1c" "GroestlCoin Signed Message:\n",   true, 0x80000011,   false,   0,     true,   8,  false, NO_CONTRACT,      true, 76067358,  true, true,   true, false,  true,"secp256k1-groestl", false, "",           true, "grs",   false, false,  true,  77429938,        true, 78792518)
```

Comment (yura-pakhuchiy) [reply include/keepkey/firmware/coins.def]: Fixed!

**Comment by yura-pakhuchiy** [comment]
@keepkeyjon Thank you for prompt review and merge, and for help with emulator! It is pleasure to contribute to KeepKey.

May I ask for signed alpha build to test GRS on a real device?

**Comment by keepkeyjon** [comment]
You're very welcome :)

We do pretty extensive testing before we sign firmwares, and since such a signed firmware would be valid for any KeepKey, we have to be very careful about what gets signed / not. (We don't have a good way to to pin it to just your device). I have run the new tests against a test device though, so we're pretty well off.

I'll ping you when the next firmware gets signed with this in it.

**Pull request merged**